File: MyList.Server/Models/UserList.cs
using System.ComponentModel.DataAnnotations;

namespace MyList.Server.Models
{
    public class UserList
    {
        [Key]
        public int Id { get; set; }
        public required string Name { get; set; }
        public string? Description { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }
        public IList<ListItem>? Items { get; set; }
    }
}

File: mylist.search-engine/src/main/kotlin/xpa/shadow/mylist/data/UserList.kt
package xpa.shadow.mylist.data

import jakarta.persistence.*
import java.io.Serializable
import java.time.Clock
import java.time.OffsetDateTime

@Entity(name = "Lists")
class UserList : Serializable {

    @Id
    var id: Int = 0
    var name: String? = null
    var description: String? = null
    var createdAt: OffsetDateTime = OffsetDateTime.now(Clock.systemUTC())
    var updatedAt: OffsetDateTime? = null

    @OneToMany(mappedBy = "list", cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
    var items: MutableList<ListItem> = mutableListOf()
}
